src/ExternalWindow.vala

```diff
@@ -81,6 +75,58 @@ public class ExternalWindowX11 : ExternalWindow, GLib.Object {
     }
 
     public void set_parent_of (Gdk.Window child_window) {
-        child_window.set_transient_for (foreign_gdk_window);
+        unowned var display = (Gdk.X11.Display) get_x11_display ();
+        unowned var x_display = display.get_xdisplay ();
+        var child_xid = ((Gdk.X11.Window) child_window).get_xid ();
+
+        x_display.set_transient_for_hint (child_xid, foreign_window);
+
+        var atom = Gdk.X11.get_xatom_by_name_for_display (display, "_NET_WM_WINDOW_TYPE_DIALOG");
+        x_display.change_property (
+            child_xid,
+            Gdk.X11.get_xatom_by_name_for_display (display, "_NET_WM_WINDOW_TYPE"),
+            X.XA_ATOM,
+            32,
+            X.PropMode.Replace,
+            (uchar[]) atom,
+            1
+        );
+    }
+}
+
+public class ExternalWindowWayland : ExternalWindow, GLib.Object {
+    private static Gdk.Display? wayland_display = null;
```

Comment:
Why do we need `wayland_display` and/or `get_wayland_display ()` at all? Both do not seem to be used in `set_parent_of`

Reply:
the display is used to check if there's an wayland server running, and raise an exception if not, the frontend don't do any validation in the `window_handle`, so it's possible to receive an "wayland:" handle while running in an x11-only session.